rfcs/0000-kuadrant_sub_components_configurations.md

```diff
@@ -0,0 +1,248 @@
+
+The Kuadrant operator should only reconcile the spec that is given. 
+This would mean if the user states the number of replicas to be used in one of the components only the replica field for that component should be reconciled.
+As the other fields would be blank at this stage, blank fields would not be reconciled to the component CR.
+By this behaviour a few things are being achieved.
+Component controllers define the defaults to be used in the components.
+Optional fields in the component CRs never get set with blank values.
+Blank values in the component CR could override the defaults of the components causing unexpected behaviour.
+Existing Kuadrant users may already have custom fields set in the component CRs.
+By only reconciling the fields set in the kuadrant CR this allows time for a user to migrate their custom configuration from the component CR to the Kuadrant CR.
+
+## Fields to reconcile
+Fields being reconcile can be classified into different groups.
+These classifications are based around the tasks a user is achieve.
+
+- Kubernetes native, setting that affect how Kubernetes handles the resource.
+- Observability, configuration settings that allow insights into how the applications are operation.
+This can be Kubernetes native or external tooling.
+- Application Settings, setting targeting the application and how it connects to external services.
+
+### Authorino Spec
+#### Kubernetes native
+- replicas
+
+#### Observability
+- healthz
+- logLevel
+- metrics
+- tracing
+
+#### Application Settings
+- eveluatorCacheSize
+- listener
+- oidcServer
+- volumes
+
+### Limitador Spec
+#### Kubernetes native
+- afffinity
+- pdb
+- replicas
+- resourceRequirements
+
+#### Application Settings
+- listener
+- storage
+
+## Fields not reconciled
+There are a number of fields in both Authorino and Limitador that are not reconciled.
+Reasons for doing this are:
+
+It is better to start with a sub set of features and expand to include more at a later date.
+Removing feature support is far harder than adding it.
+
+There are four classifications the unreconciled fields fail into.
+- Deprecated, fields that are deprecated and/or have plans to be removed from the spec in the future.
+- Unsupported, the features would have hard coded or expected defaults in the Kuadrant operator. 
+Work would be required to all the custom configurations options.
+- Dev/Testing focused, features that should only be used during development & testing and not recommended for production.
+The defaults would for the fields are the production recommendations.
+- Reconciled by others, this mostly affects Limitador as the deployment configuration and runtime configuration are in the same CR.
+In the case of Kuadrant the runtime configuration for Limitador is added via the RateLimitingPolicy CR.
+
+### Authorino Spec
+
+#### Unsupported
+- clusterWide
+- authConfigLabelSelectors
+- secrtLabelSelectors
+
+#### Dev/Testing focused
```

Comment:
I would actually include those as well but with a clear indication that it should not be used for production. It might make debugging or trying out newer Authorino versions simpler.

Reply:
I am in two minds about this, yes it would make using newer version of Authorino easier but also people don't read the documentation so how do you give a clear indication it should not be used in production. By making the route to make such changes harder it is an indication that a user should not being doing it. At no point am I suggesting that we don't allow the setting of these fields but just that it should be done it the sub operator directly. 

Take the example  of the `image` field. it allows the setting of a custom Authorino image. If this is set in the Kuadrant CR the custom image will be set during the installation of Kuadrant. When the image is set in the Authorino CR the Authpolicies have already being created. This would be a very deliberate act need to use a custom image.  

If the customer is using GitOps to manage the resources setting a custom image in the Kuadrant CR is very simple but also setting it the image in Authorino would be possible, it needs a more advance set up which should be an indication that this might not be the correct way. 

When it comes to a customer using custom images and pull policies I would suspect they are also the kind of customer that does not use auto update and well use to having extra steps to ensure the their configuration.

Reply:
And will we allow setting things in suboperator, or is it expected that Kuadrant reconciles back all of the changes?

Reply:
Yes, if the field can not be set in the Kuadrant CR the kuadrant-operator should not reconcile back any changes to that field. This allow the configuration of the test/dev features directly on the suboperator. A side affect of this would mean any new feature in the suboperators do not need to be in the Kuadrant CR for the user to make use of the feature. This would be like a soft launch.

Thinking about what this means for upgrades. If the kuadrant-operator reverts the configuration on the suboperators to the configuration in the Kuadrant CR this could cause issues for the end users. But also can be documented as an upgrade step/requirement. I believe this would be an acceptable step.

Reply:
Ok, that makes sense to me